designer/server/src/main/scala/pl/touk/nussknacker/ui/Error.scala

```diff
@@ -0,0 +1,31 @@
+package pl.touk.nussknacker.ui
+
+import akka.http.scaladsl.model.{StatusCode, StatusCodes}
+
+
+object Error {
+
+  type XError[A] = Either[Error, A]
+
+}
+
+trait Error {
+  def getMessage: String
+  val statusCode: Option[StatusCode] = None
```

Comment:
Is `Option` really required here? All errors we declared have `statusCode` assigned. Or maybe we want to allow creating new errors in new places? If so forget about my 2 comments :)

Reply:
We want to allow creating new errors in new places